mapbox-lint/src/main/java/com/mapbox/maps/lint/LifecycleMethodDetector.kt

```diff
@@ -0,0 +1,46 @@
+package com.mapbox.maps.lint
+
+import com.android.tools.lint.detector.api.*
+import com.intellij.psi.PsiMethod
+import org.jetbrains.uast.UCallExpression
+
+/**
+ * Lint detector for lifecycle events
+ */
+class LifecycleMethodDetector : Detector(), Detector.UastScanner {
+
+  override fun getApplicableMethodNames()= listOf(ON_START, ON_STOP, ON_DESTROY, ON_LOW_MEMORY)
+
+  override fun visitMethodCall(context: JavaContext, node: UCallExpression, method: PsiMethod) {
+    if(context.evaluator.isMemberInClass(method, "com.mapbox.maps.MapView")){
+      context.report(
+        ISSUE, node, context.getLocation(node),
+        REPORT_MESSAGE
+      )
+    }
+  }
+
+  companion object Issues {
+    private const val ON_START = "onStart"
+    private const val ON_STOP = "onStop"
+    private const val ON_DESTROY = "onDestroy"
+    private const val ON_LOW_MEMORY = "onLowMemory"
+    private const val ISSUE_ID = "Lifecycle"
+    private const val BRIEF_DESCRIPTION = "You have redundant methods call"
+    private const val REPORT_MESSAGE =
+      "No need to invoke onStart/onStop/onDestroy/onLowMemory with Mapbox Lifecycle Plugin."
```

Comment:
```suggestion
      "No need to invoke onStart/onStop/onDestroy/onLowMemory explicitly when Mapbox Lifecycle Plugin is added as dependency."
```

Reply:
I think we could mention that the lifecycle plugin is a transitive dependency of Maps SDK by default, and we should mention the exception when appcompact is not used or the version of appcompact is below 13.x

Reply:
By the way can we also validate the appcompact version using lint?

Reply:
I think we can't validate appcompact version here.

Reply:
For now we don't have to validate the appcompact version, but would be really good to add some instruction here as it would be required to overwrite manually if the user uses appcompact before 1.3.0